File: BusinessLogic/MoveCommand.cs
namespace CocaineCartels.BusinessLogic
{
    public class MoveCommand : Command
    {
        public MoveCommand(Unit unit, Cell from)
            : base(unit)
        {
            From = from;
        }

        internal readonly Cell From;

        public Hex FromHex => From.Hex;
    }
}

File: WebApplication/Client/MoveCommand.ts
module CocaineCartels {
    "use strict";

    export class MoveCommand extends Command {
        constructor(
            public unit: Unit,
            public from: Cell,
            public to: Cell
        ) {
            super(CommandType.MoveCommand, unit);

            if (unit.cell === null && unit.placeCommand === null) {
                throw "Can only assign move commands to units that are placed on a cell or has a place command.";
            }
        }

        private _color: string = undefined;

        public get color() {
            if (this._color === undefined) {
                this._color = this.unit.player.color;
            }

            return this._color;
        }
    }
}
